service/src/main/java/bio/terra/workspace/service/iam/SamRethrow.java

```diff
@@ -0,0 +1,60 @@
+package bio.terra.workspace.service.iam;
+
+import bio.terra.common.sam.exception.SamExceptionFactory;
+
+/**
+ * When we use Sam in flights, we want to let InterruptedException fly and be caught by Stairway.
+ * However, when we use Sam outside of flights, we want to propagate the Interrupted exception.
+ * These static methods perform that propagation.
+ */
+public class SamRethrow {
+
+  /**
+   * For use with onInterrupted methods.
+   *
+   * @param <R>
+   */
+  @FunctionalInterface
+  public interface InterruptedSupplier<R> {
+    R apply() throws InterruptedException;
+  }
+
+  /** For use with onInterrupted. */
+  public interface VoidInterruptedSupplier {
```

Comment:
I'd call this `InterruptedRunner` to be consistent with the `Runnable` convention.